### PR TITLE
Change dto for page blog documents to access fields in safe manner

### PR DESCRIPTION
**Pull request by artem-shelkovnikov**
Minor change to replace unsafe object field access with safe one - we've received reports with errors coming from these lines because some intermediate objects were None:

```
Traceback (most recent call last):
  File "/app/connectors/sources/confluence.py", line 1154, in _page_blog_coro
    async for document, attachment_count, space_key, permissions, restrictions in self.fetch_documents(
  File "/app/connectors/sources/confluence.py", line 904, in fetch_documents
    "author": document["history"]["createdBy"][self.authorkey],
KeyError: 'publicName'
[FMWK][18:00:15][ERROR] Exception found for task Task-132
Traceback (most recent call last):
  File "/app/connectors/sources/confluence.py", line 1154, in _page_blog_coro
    async for document, attachment_count, space_key, permissions, restrictions in self.fetch_documents(
  File "/app/connectors/sources/confluence.py", line 904, in fetch_documents
    "author": document["history"]["createdBy"][self.authorkey],
KeyError: 'publicName'
```

#### Pre-Review Checklist
- [x] this PR does NOT contain credentials of any kind, such as API keys or username/passwords (double check `config.yml.example`)
- [x] this PR has a meaningful title
- [ ] this PR links to all relevant github issues that it fixes or partially addresses
- [ ] if there is no GH issue, please create it. Each PR should have a link to an issue
- [x] this PR has a thorough description
- [x] Covered the changes with automated tests
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)

## Release Note

Fix Confluence connector not downloading some blog post documents due to unexpected response format.

### CODE REVIEW

**Review by jedrazb — APPROVED** [review]
Nice! 🚢

File: connectors/sources/confluence.py [hunk 1]
```diff
@@ -908,13 +908,19 @@ async def fetch_documents(self, api_query):
...
+                "space": nested_get_from_dict(document, ["space", "name"]),
+                "body": html_to_text(
+                    nested_get_from_dict(document, ["body", "storage", "value"])
+                ),
                 "url": document_url,
-                "author": document["history"]["createdBy"][self.authorkey],
+                "author": nested_get_from_dict(
+                    document, ["history", "createdBy", self.authorkey]
+                ),
                 "createdDate": document["history"]["createdDate"],
```
Comment (jedrazb) [inline connectors/sources/confluence.py:924]: should we update this as well?

Comment (artem-shelkovnikov) [reply connectors/sources/confluence.py:924]: True! Idk how I've missed it

**Review by jedrazb — APPROVED** [review]

**Comment by artem-shelkovnikov** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.15|[](https://github.com/elastic/connectors/pull/2988)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**